### PR TITLE
Updates to default training procedures

### PR DESCRIPTION
**Pull request by wisecornelius**
## Proposed changes

Update default training behaviour of `run_model`.

- Fix seed in examples
- use gradient clipping
- track gradient norm
- use AMP

## Types of changes

What types of changes does your code introduce to neural-lifetimes?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] Description above provides context of the change
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions

### CODE REVIEW

File: examples/eventsprofiles_btyd.py [hunk 1]
```diff
@@ -23,79 +25,82 @@
 START_TOKEN_DISCR = "StartToken"
 COLS = eventsprofiles_datamodel.target_cols + eventsprofiles_datamodel.cont_feat + eventsprofiles_datamodel.discr_feat
 
-data_dir = str(Path(__file__).parent.absolute())
+if __name__ == "__main__":
+    pl.seed_everything(9473)
```
Comment (pereplanellTW) [inline examples/eventsprofiles_btyd.py:29]: Any reason for this seed? Is there an easy way to remember it?

**Review by pereplanellTW — APPROVED** [review]

File: examples/eventsprofiles_btyd.py (hunk 1 above)
Comment (wisecornelius) [reply examples/eventsprofiles_btyd.py:29]: it spells WISE :)
